python_modules/libraries/dagster-aws/dagster_aws/emr/emr_eks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,569 @@
+                "This can be useful, for example, in case of git submodules."
+            ),
+            default_value=[],
+        ),
+        "working_directory_override": Field(
+            StringSource,
+            is_required=False,
+            description=(
+                "EMR on EKS assumes the entry point for the job to be located in ``. "
+                "If the user code is installed somewhere else in the container, the default "
+                "value needs to be overridden."
+            ),
+        ),
+        "log4j_conf": Field(Permissive(), default_value={}, is_required=False),
+    },
+)
+
+
+@resource(config_schema=EMR_EKS_CONFIG_SCHEMA)
+def emr_eks_pyspark_resource(context):
+    spark_config = context.resource_config.get("spark_config")
+
+    return EmrEksPySparkResource(
+        cluster_id=context.resource_config.get("cluster_id"),
+        container_image=context.resource_config.get("container_image"),
+        deploy_local_job_package=context.resource_config.get("deploy_local_job_package"),
+        emr_release_label=context.resource_config.get("emr_release_label"),
+        job_role_arn=context.resource_config.get("job_role_arn"),
+        log_group_name=context.resource_config.get("log_group_name"),
+        log4j_conf=context.resource_config.get("log4j_conf"),
+        region_name=context.resource_config.get("region_name"),
+        staging_bucket=context.resource_config.get("staging_bucket"),
+        staging_prefix=context.resource_config.get("staging_prefix"),
+        spark_config=spark_config,
+        ephemeral_instance=context.instance.is_ephemeral,
+        additional_relative_paths=context.resource_config.get("additional_relative_paths"),
+        working_directory_override=context.resource_config.get("working_directory_override"),
+    )
+
+
+class EmrEksPySparkResource(PySparkResource, EmrPySparkStepLauncherBase):
+    """
+    A PySpark resource where code is executed on a remote
+    EMR on EKS cluster.
+
+    Launching Steps on Remote Clusters
+
+    When this resource implementation is used as a `PySparkResource`,
+    a job run on EMR on EKS will be started, and the step will run there.
+    This works by making the resource also inherit from `StepLauncher`,
+    and implement `launch_step` to serialize the definition of the step,
+    run it on the remote cluster.
+
+    Dagster operations produce events (e.g., when a step finishes, or when
+    an output is produced). The events produced by the operation running
+    on EMR on EKS in the Spark driver thus need to be serialized, sent
+    back to the Dagster instance, and deserialized so that Dagster is
+    aware of what happened in the remote cluster. Dagster events are not
+    JSON serialized, and have to be pickled. As these events are small,
+    the following procedure is used:
+
+    - In the Spark driver, as part of the job entrypoint (`job_main.py`),
```

Comment:
nit: is this still the right name for the entrypoint? We can also drop the filename from the doc